### ISSUE TITLE
IRQ_HEADER_ERROR should raise the RxError event in RadioBgIrqProcess (Radio.cpp)

### ISSUE THREAD

**Issue by JeromeBriot** [beegee-tokyo/SX126x-Arduino#110]
Hello,

The block of code below in RadioBgIrqProcess (Radio.cpp):

```cpp
if ((irqRegs & IRQ_HEADER_ERROR) == IRQ_HEADER_ERROR)
{
	LOG_LIB("RADIO", "RadioIrqProcess => IRQ_HEADER_ERROR");

	TimerStop(&RxTimeoutTimer);
	if (RxContinuous == false)
	{
		//!< Update operating mode state to a value lower than \ref MODE_STDBY_XOSC
		SX126xSetOperatingMode(MODE_STDBY_RC);
	}
	if ((RadioEvents != NULL) && (RadioEvents->RxTimeout != NULL))
	{
		RadioEvents->RxTimeout();
	}
}
```

Should be:

```cpp
if ((irqRegs & IRQ_HEADER_ERROR) == IRQ_HEADER_ERROR)
{
	LOG_LIB("RADIO", "RadioIrqProcess => IRQ_HEADER_ERROR");

	TimerStop(&RxTimeoutTimer);
	if (RxContinuous == false)
	{
		//!< Update operating mode state to a value lower than \ref MODE_STDBY_XOSC
		SX126xSetOperatingMode(MODE_STDBY_RC);
	}
	if ((RadioEvents != NULL) && (RadioEvents->RxTimeout != NULL))
	{
		RadioEvents->RxError();
	}
}
```

What do you think?

**Comment by beegee-tokyo** [comment]
Yes, that make sense. With a small change:
```cpp
if ((RadioEvents != NULL) && (RadioEvents->RxTimeout != NULL))
```
should be 
```cpp
if ((RadioEvents != NULL) && (RadioEvents->RxError!= NULL))
```
Otherwise it would lead to a crash if RxError function is not set.

**Comment by beegee-tokyo** [comment]
Implemented in V2.0.22
